### ISSUE TITLE
Improve code coverage for System.IO.FileSystem

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#927]
According to the current code coverage report, this library is only at 44%.  Our general goal is closer to 90%, though this will depend on the library.  We should investigate to see whether the current coverage level is appropriate, and if not, improve our test suites to provide better coverage.

The low number is certainly due to missing part of the test suite.  We should reevaulate once #843 is completed.

**Comment by VanJanssen** [comment]
I would like to contribute to this issue. What are the naming conventions for test cases and test files?

**Comment by stephentoub** [comment]
@ericstj?

**Comment by ericstj** [comment]
Naming convention is folder per class, filename per member.  I have a branch where I have written a bunch of tests for File and I think @FiveTimesTheFun is also working on some.

**Comment by VanJanssen** [comment]
Thanks for the info. In my enthusiasm I commented on this issue before realising it should be put on hold until #843 is completed. I will look through the up for grabs issues to see where I can best contribute.

**Comment by stephentoub** [comment]
Thanks, @VanJanssen.

**Comment by ericstj** [comment]
@FiveTimesTheFun would you consider this resolved with your change?

**Comment by ianhays** [comment]
What is the state of this issue? I'm planning on working towards #2227 to solve the intermittent failures on the FileSystem tests but don't want to do anything that's already been resolved elsewhere.

@ericstj is your branch with changes to File tests mentioned in your above comment merged with master already?

**Comment by stephentoub** [comment]
> What is the state of this issue?

You can see a current report at http://dotnet-ci.cloudapp.net/job/dotnet_corefx_coverage_windows/lastStableBuild/Code_Coverage_Report/ (or generate one locally using the ```/p:Coverage=true``` build flag).

The current line coverage is 83% and the current branch coverage is 73%.  While it'd be nice to boost those higher, that's more a side-effect than a goal: it's less about the number and more about making sure we have the right set of tests in place.  Are we exercising all of the right corner cases based on using a good and varied enough set of input data?  Are we exercising all of the error paths?  Are we appropriately representing all of the various use cases for these APIs? Etc.

From a cross-platform perspective, we'd really like to be able to use the tests effectively as an oracle, or as test-driven development, depending on how you want to think about it: if we had a perfect set of tests exercising all possible scenarios, then we should be able to get a functionally-conforming non-Windows implementation by driving the number of failures on such a platform to 0.

**Comment by ianhays** [comment]
After my numerous reworks and expansions to the FileSystem tests (#2390, #2433, #2512), I'm going to close this. Coverage for all public-facing APIs are at or near their testable maximums with a few exceptions contingent on solutions to issues with the build system (#1220, #1221).
